### ISSUE TITLE
compilation fails with 'error: "glXSwapIntervalEXT' was not declared in this scope"

### ISSUE THREAD

**Issue by madjestic** [godotengine/godot#5060]
Hey, guys

I am getting a compile error, it looks like a missing dependency.. I am running GLEW 1.13.0, glut 2.7.0.1 and glu 9.x.

**Operating system or device - Godot version:**
Linux 4.4.6-gentoo x86_64 Intel(R) Core(TM) i5-3230M CPU @ 2.60GHz GenuineIntel GNU/Linux

**Issue description** (what happened, and what was expected):
compilation fails

**Steps to reproduce:**
$ scons platform=x11
...
error

```bash
$ scons platform=x11
scons: Reading SConscript files ...
Enabling ALSA
Enabling PulseAudio
('translations: ', ['/home/madjestic/Projects/godot/tools/translations/ru.po', '/home/madjestic/Projects/godot/tools/translations/zh_HK.po', '/home/madjestic/Projects/godot/tools/translations/pt_BR.po', '/home/madjestic/Projects/godot/tools/translations/ko.po', '/home/madjestic/Projects/godot/tools/translations/it.po', '/home/madjestic/Projects/godot/tools/translations/de.po', '/home/madjestic/Projects/godot/tools/translations/fr.po', '/home/madjestic/Projects/godot/tools/translations/zh_CN.po'])
('fonts: ', ['/home/madjestic/Projects/godot/tools/editor_fonts/DroidSansThai.ttf', '/home/madjestic/Projects/godot/tools/editor_fonts/DroidSans.ttf', '/home/madjestic/Projects/godot/tools/editor_fonts/DroidSansArabic.ttf', '/home/madjestic/Projects/godot/tools/editor_fonts/DroidSansHebrew.ttf', '/home/madjestic/Projects/godot/tools/editor_fonts/DroidSansJapanese.ttf', '/home/madjestic/Projects/godot/tools/editor_fonts/DroidSansFallback.ttf', ['/home/madjestic/Projects/godot/tools/editor_fonts/source_code_pro.otf']])
scons: done reading SConscript files.
scons: Building targets ...
g++ -o platform/x11/context_gl_x11.x11.tools.64.o -c -g2 -Wall -DDEBUG_ENABLED -DDEBUG_MEMORY_ENABLED -DFREETYPE_ENABLED -DOC_X86_ASM -DDEBUG_MEMORY_ALLOC -DSCI_NAMESPACE -DOPENGL_ENABLED -DGLEW_ENABLED -DALSA_ENABLED -DJOYDEV_ENABLED -DPULSEAUDIO_ENABLED -DX11_ENABLED -DUNIX_ENABLED -DGLES2_ENABLED -DGLES_OVER_GL -DMUSEPACK_ENABLED -DSQUISH_ENABLED -DVORBIS_ENABLED -DOPUS_ENABLED -DTHEORA_ENABLED -DTHEORALIB_ENABLED -DPNG_ENABLED -DDDS_ENABLED -DPVR_ENABLED -DJPG_ENABLED -DWEBP_ENABLED -DSPEEX_ENABLED -DTOOLS_ENABLED -DGDSCRIPT_ENABLED -DMINIZIP_ENABLED -DXML_ENABLED -DETC1_ENABLED -D_REENTRANT -Icore -Icore/math -Itools -Idrivers -I. -Iplatform/x11 -I/usr/include/freetype2 platform/x11/context_gl_x11.cpp
platform/x11/context_gl_x11.cpp: In member function 'virtual void ContextGL_X11::set_use_vsync(bool)':
platform/x11/context_gl_x11.cpp:181:53: error: 'glXSwapIntervalEXT' was not declared in this scope
  glXSwapIntervalEXT(x11_display, drawable, p_use?1:0);
                                                     ^
scons: *** [platform/x11/context_gl_x11.x11.tools.64.o] Error 1
scons: building terminated because of errors.
```

**Comment by vnen** [comment]
According to commit 2420e46b449f4c8acdfe48c765ea52fc3e860de7 it was expected to not work. Travis failed too: https://travis-ci.org/godotengine/godot/jobs/135472746

It seems @punto- is looking to this right now though.

**Comment by suptoasty** [comment]
GLX instead of glX fixed this problem for me.

**Comment by punto-** [comment]
I get no difference with GLX, defining GLX_GLXEXT_PROTOTYPES provides the
prototype for the function, but it doesn't link, I think glew has to fetch
the extension, but I don't know how to request that

**Comment by akien-mga** [comment]
Fixed by #5067.
